lib/api/route/messages.dart

```diff
@@ -263,3 +263,60 @@ class UploadFileResult {
 
   Map<String, dynamic> toJson() => _$UploadFileResultToJson(this);
 }
+
+/// https://zulip.com/api/add-reaction
+Future<AddReactionResult> addReaction(ApiConnection connection, {
+  required int messageId,
+  required ReactionType reactionType,
+  required String emojiCode,
+  required String emojiName,
+}) {
+  return connection.post('addReaction', AddReactionResult.fromJson, 'messages/$messageId/reactions', {
+    'emoji_name': RawParameter(emojiName),
+    'emoji_code': RawParameter(emojiCode),
+    'reaction_type': RawParameter(switch (reactionType) {
+      ReactionType.unicodeEmoji => 'unicode_emoji',
+      ReactionType.realmEmoji => 'realm_emoji',
+      ReactionType.zulipExtraEmoji => 'zulip_extra_emoji',
```

Comment:
Hmm, awkward to have to repeat this switch. Can we get json_serializable to do this for us?

Yeah, looks like we can say this:
```dart
  zulipExtraEmoji;

  String toJson() => _$ReactionTypeEnumMap[this]!;
}
```

and then I think this can become `RawParameter(reactionType.toJson())`.